cmd/syncComparator.go

```diff
@@ -164,33 +165,83 @@ func (f *syncDestinationComparator) FinalizeTargetDirectory(folderMap *objectInd
 	// 2. Does only metadata need to be sync'ed?
 	// 3. Object is not changed since last sync, no need to update.
 	//
+	// Note: For child folder we update its properties in it's finalizer.
+	// Child folder will be updated when they are enumerated. f.e. following directory structure present on source
+	// 1. [dir1][dir2]
+	// 2. [dir1][file1]
+	// 3. [dir2][file2]
+	// Its object indexer map entries will be following :-
+	// 1. [dir1]["."]
+	// 2. [dir1][dir2]
+	// 3. [dir1][file1]
+	// 4. [dir2]["."]
+	// 5. [dir2][file2]
+	// For dir1 finalizer we update folder properties of dir1 and file1. On dir2 finalizer we do dir2 folder properties updation and file2.
+	// As finalizer should be called for each folder, and it make sense to update its properties after creating its child files.
+	//
 
 	// For FinalizeAll flag true, we know target traverser done. Now the files left in ObjectIndexerMap are the new ones.
 	// So, we don't do any ctime and mtime comparsion.
 	if finalizeAll {
```

Comment:
for now, to prevent the above  f.copyTransferScheduler(storedObject) call for a folder, you can add the following code above this

          // TODO: Remove this code.
          //       Till we add support for updating directory attributes, we prevent this directory
          //       creation, else this clashes with the directory creation at the end of FinalizeTargetDirectory().
          //       Once we add support for directory attributes updation, change that call to cause
          //       directory attributes updation and not directory creation and remove this code.
          //
          if storedObject.entityType == common.EEntityType.Folder() {
              dataChange = false
          }